src/ejabberd_sm.erl

```diff
@@ -364,17 +366,19 @@ unset_presence(Acc, SID, User, Server, Resource, Status, Info) ->
                         jid:resourceprep(Resource), Status]).
 
 
--spec close_session_unset_presence(SID, User, Server, Resource, Status, Reason) -> ok when
+-spec close_session_unset_presence(Acc, SID, User, Server, Resource, Status, Reason) -> Acc1 when
+      Acc :: mongoose_acc:t(),
       SID :: 'undefined' | sid(),
       User :: jid:user(),
       Server :: jid:server(),
       Resource :: jid:resource(),
       Status :: any(),
-      Reason :: close_reason().
-close_session_unset_presence(SID, User, Server, Resource, Status, Reason) ->
-    close_session(SID, User, Server, Resource, Reason),
+      Reason :: close_reason(),
+      Acc1 :: mongoose_acc:t().
+close_session_unset_presence(Acc, SID, User, Server, Resource, Status, Reason) ->
+    close_session(Acc, SID, User, Server, Resource, Reason),
     LServer = jid:nameprep(Server),
-    ejabberd_hooks:run(unset_presence_hook, LServer,
+    ejabberd_hooks:run_fold(unset_presence_hook, LServer, Acc,
```

Comment:
Do we want to pass the `Acc1` we get from calling `close_session/6` two lines above?

Reply:
It's not strictly necessary, but it makes it uglily inconsistent not to. Thanks for the catch!